lib/PipelineC/PipelineC.cpp

```diff
@@ -213,11 +221,19 @@ bool rp_manager_produce_targets(rp_manager *manager,
     Targets[container->second->name()].push_back(*targets[I]);
 
   auto Error = manager->getRunner().run(step->getName(), Targets);
-  if (not Error)
-    return true;
+  if (Error) {
+    llvm::consumeError(std::move(Error));
+    return nullptr;
+  }
 
-  llvm::consumeError(std::move(Error));
-  return false;
+  std::string Out;
+  llvm::raw_string_ostream Serialized(Out);
+  const auto &ToFilter = Targets[container->second->name()];
+  const auto &Cloned = container->second->cloneFiltered(ToFilter);
+  llvm::cantFail(Cloned->serialize(Serialized));
+  Serialized.flush();
+
+  return copy_string(Out);
```

Comment:
OK, still, it could be done. This API might transfer biiiig chunks of data.
You could have an heap allocated buffer, have the stream write in there, and then steal the ownership of data and pass it to the caller.

Reply:
yeah, the correct solution would be to write a custom output stream that writes in a buffer. i would delay this until it becomes a performance bottleneck toh

Reply:
Works! My original comment was "add a TODO about this in the code".

Reply:
`TODO`s are typically things that never get done. But once you identify the problem, you look at the code and you find a suggestion on how to fix a problem we foresaw. Imagine that the person finding the comment might not be you,